siege_utilities/engines/dataframe_engine.py

```
@@ -936,10 +956,18 @@ def read_spatial(self, path: str, *, crs: Optional[str] = None, **kwargs: Any) -
     def spatial_join(self, left, right, how="inner", predicate="intersects",
                      *, left_geom="geometry", right_geom="geometry"):
         self._ensure_sedona()
+        from siege_utilities.core.sql_safety import (
+            validate_sql_identifier_in as validate_identifier_in,
+            validate_sql_identifier as validate_identifier,
+        )
+        validate_identifier_in(left_geom, left.columns, label="left geometry column")
+        validate_identifier_in(right_geom, right.columns, label="right geometry column")
+        validate_identifier(predicate, label="spatial predicate", allow_dotted=False)
+        if how.lower() not in ("inner", "left", "right", "outer", "full"):
+            raise ValueError(f"unsupported join type {how!r}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)